pkg/console/controllers/advancedclustermanagement/managedclusteroauthclients/controller.go

```diff
@@ -0,0 +1,200 @@
+		managedClusterClient: managedClusterClient,
+		oauthClient:          oauthClient,
+		workClient:           workClient,
+	}
+
+	configV1Informers := configInformer.Config().V1()
+
+	return factory.New().
+		WithFilteredEventsInformers( // configs
+			util.IncludeNamesFilter(api.ConfigResourceName),
+			configV1Informers.Consoles().Informer(),
+			operatorConfigInformer.Informer(),
+		).
+		ResyncEvery(1*time.Minute).
+		WithSync(ctrl.Sync).
+		ToController("ManagedClusterOAuthClientController", recorder.WithComponentSuffix("managed-cluster-oauth-client-controller"))
+}
+
+func (c *Controller) Sync(ctx context.Context, controllerContext factory.SyncContext) error {
+	operatorConfig, err := c.operatorConfigClient.Get(ctx, api.ConfigResourceName, metav1.GetOptions{})
+	if err != nil {
+		return err
+	}
+
+	switch operatorConfig.Spec.ManagementState {
+	case operatorv1.Managed:
+		klog.V(4).Info("console-operator is in a managed state: syncing managed cluster oauth clients")
+	case operatorv1.Unmanaged:
+		klog.V(4).Info("console-operator is in an unmanaged state: skipping managed cluster oauth client sync")
+		return nil
+	case operatorv1.Removed:
+		klog.V(4).Infof("console-operator is in a removed state: deleting managed cluster oauth clients")
+		return c.remove(ctx)
+	default:
+		return fmt.Errorf("unknown state: %v", operatorConfig.Spec.ManagementState)
+	}
+
+	statusHandler := status.NewStatusHandler(c.operatorClient)
+
+	// Get the local OAuthClient. If this fails, do not proceed. We can't create OAuth clients on
+	// managed clusters without the local client.
+	localOAuthClient, reason, err := c.GetLocalOAuthClient(ctx)
+	statusHandler.AddConditions(status.HandleProgressingOrDegraded(ConditionPrefix, reason, err))
+	if err != nil {
+		return c.remove(ctx)
+	}
+
+	managedClusterList, reason, err := c.SyncManagedClusters(ctx)
+	statusHandler.AddConditions(status.HandleProgressingOrDegraded(ConditionPrefix, reason, err))
+	if err != nil || len(managedClusterList.Items) == 0 {
+		return c.remove(ctx)
+	}
+
+	// Create ManifestWorks which will create oauth clients on each managed cluster
+	errs := []error{}
+	for _, managedCluster := range managedClusterList.Items {
+		validCluster, validClusterVersion := util.IsSupportedMangedCluster(managedCluster)
+		if !validCluster || !validClusterVersion {
+			klog.V(4).Infof("Skipping oauth client ManifestWork on managed cluster %s, cluster is not supported./n", managedCluster.Name)
+			continue
+		}
+
+		reason, err = c.SyncOauthClientManifestWork(
+			ctx,
+			operatorConfig,
+			managedCluster,
+			localOAuthClient.Secret,
+			localOAuthClient.RedirectURIs,
+		)
+		statusHandler.AddConditions(status.HandleProgressingOrDegraded(ConditionPrefix, reason, err))
+		if err != nil {
+			klog.V(4).Infof("Error syncing OAuthClient ManifestWork for managed cluster %s: %v./n", managedCluster.Name, err)
+			errs = append(errs, err)
+		}
+	}
+
+	err = utilerrors.NewAggregate(errs)
```

Comment:
Lets rather move the `statusHandler.AddConditions` here and try to aggregate the `reason` as well, otherwise only the last failing sync will be surfaced by in the condition. We should surface all of them